pkg/apiserver/user/sqlauth/sqlauth.go

```diff
@@ -50,6 +65,29 @@ func (a *Authenticator) Authenticate(f user.AuthenticateForm) (*utils.SessionUse
 	}
 	defer utils.CloseTiDBConnection(db) //nolint:errcheck
 
+	// Check dashboard privileges
+	// 1. Check whether TiDB SEM is enabled
+	resData, err := a.tidbClient.SendGetRequest("/config")
+	if err != nil {
+		return nil, user.ErrSignInOther.WrapWithNoMessage(err)
+	}
```

Comment:
Will this lead to confusing error messages when TiDB is down?

Reply:
Let me check.

Reply:
Add `time.Sleep(10*time.Second)` before requesting `/config`, and kill the TiDB after clicking the `Sign In` button, it reports `Sign in failed: authenticate failed, cause: error.api.user.signin.other: error.tidb.no_alive_tidb` error message, I think it is clear enough.

![image](https://user-images.githubusercontent.com/1284531/131768812-0b54ee92-1271-44d8-9f32-0b5875abe88c.png)

Reply:
Btw, in the RP #992 , this part logic is refactored (extracted for reusing), the `error.tidb.no_alive_tidb` error is returned directly, finally it only show `error.tidb.no_alive_tidb` error here.